meson.build

```diff
@@ -234,7 +234,8 @@ endif
 
 # Optional dependency on GObject-Introspection; if GObject is disabled
 # then we don't build introspection data either
-build_gir = build_gobject and get_option('introspection') and get_option('default_library') == 'shared'
+gir = find_program('g-ir-scanner', required : get_option('introspection'))
+build_gir = build_gobject and gir.found() and (not meson.is_cross_build() or get_option('introspection').enabled())
```

Comment:
This line is really hard to read. Additionally, `is_cross_build()` is not necessary, because anybody cross-compiling Graphene either explicitly disables introspection *or* have a native g-i build that they can use.

Reply:
The whole point here is to not have to explicitly disable when it's set to auto. That's the same line copied in many projects, I don't think graphene is different.

Reply:
It's still an unreadable jumble of conditions that is hard to read.

Reply:
IIRC native g-i is not enough since it compiles and run code which only works if you have set an exe wrapper in your cross file. With Meson >= 0.55 I think the condition `not meson.is_cross_build() or get_option('introspection').enabled()` could be replaced by `can_run_host_binaries()`.

Reply:
We can bump up the version of Meson to 0.55/0.56, if it means improving the legibility/correctness of this block. Especially if it avoids a new merge request (or, worse, downstream patch) from projects like Yocto to remove the `is_cross_build()` check.

In any case, and to make it even more painfully clear: my objection was that the line is hard to parse from a maintainer standpoint.

I don't immediately remember the Meson operator precedence, and I don't want to. Here we have four conditions, and one of them is negated? Or maybe both the ones in the parenthesis? The length of the line doesn't make it easier to read, either.

Can you please expand this to an `if...endif` block instead of golfing it? Something like:

```
build_gir = build_gobject
if build_gir
  if not get_option('introspection').enabled() or not gir.found()
    build_gir = false
  endif
  if not meson.can_run_host_binaries()
    build_gir = false
  endif
endif
```

Reply:
Ok, I splitted the condition to make it more readable.

Reply:
@ebassi does it look good to you now?

Reply:
Yes, this looks a lot better, thanks.